### PR TITLE
Add regex capability to select files for sftp source connection

### PR DESCRIPTION
**Pull request by ejaxon**
This PR adds the ability to use a regular expression to match files on an sftp get, with additional parameters to filter and select a final single file for download.

It also updates local.py to allow it to be called with a file argument, rather than having the local configuration embedded in the code.

### CODE REVIEW

**Review by bitslayer42 — APPROVED** [review]

**Review by CameronHenshaw — APPROVED** [review]

**Comment by ejaxon** [comment]
BTW once approved, one of you will need to merge - I don't have write access.

**Pull request merged**
